src/memoize.ts

```diff
@@ -38,8 +38,12 @@ function memoize<
 >(f: F, resolver?: R) {
   const memoized = (...args: Parameters<typeof f>): ReturnType<F> => {
     const key = typeof resolver === "function" ? resolver(...args) : args[0];
-    const cache = memoized.cache;
-
+    const _self = memoized as any;
+    if (_self.cache === undefined) {
+      _self.cache =
+        key != null && typeof key === "object" ? new WeakMap() : new Map();
+    }
+    const { cache } = _self;
```

Comment:
`weakMap` does not support the keys(), values(), entries() methods. Therefore, since you can access the map directly from the outside, it would be safer to have a separate interface to manipulate the cache.

Reply:
So, you're suggesting to restrict the usage of WeakMap as a type for cache and just use Map instead, right?

Reply:
No, how about providing an abstracted method?

Reply:
Can you please write it in code and suggest it?

Reply:
> Therefore, since you can access the map directly from the outside

I think the sentence above that I wrote before is wrong. I think I'm mistaken. I don't think it's a problem because _self is only used internally. 🙏